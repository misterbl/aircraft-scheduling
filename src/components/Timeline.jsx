import React, { PureComponent } from "react";
import { DAYTOSECONDS } from "../const";
import { calculatePercentage } from "../utils";

class Timeline extends PureComponent {
  startIdl = () => {
    const { flyingTimesList } = this.props;
    if (flyingTimesList[0]) return flyingTimesList[0].departuretime;
  };

  render() {
    const { flyingTimesList } = this.props;
    const startIdl =
      flyingTimesList[0] &&
      calculatePercentage(flyingTimesList[0].departuretime);

    const endIdl =
      flyingTimesList[-1] &&
      calculatePercentage(DAYTOSECONDS - flyingTimesList[-1].arrivaltime);

    return (
      <div className="progress">
        <div
          key={startIdl}
          role="progressbar"
          style={{ width: `${startIdl}%` }}
          aria-valuenow={startIdl}
          aria-valuemin="0"
          aria-valuemax="100"
        />
        {flyingTimesList.map(flight => (
          <>
            {flight && (
              <div
                key={flight.timePercentage}
                className={`progress-bar ${
                  flight.turnaround ? "bg-danger" : "bg-success"
                }`}
                role="progressbar"
                style={{ width: `${flight.timePercentage}%` }}
                aria-valuenow={flight.timePercentage}
                aria-valuemin="0"
                aria-valuemax="100"
              />
            )}
            <div
              key={endIdl}
              role="progressbar"
              style={{ width: `${endIdl}%` }}
              aria-valuenow={endIdl}
              aria-valuemin="0"
              aria-valuemax="100"
            />
          </>
        ))}
      </div>
    );
  }
}
export default Timeline;
