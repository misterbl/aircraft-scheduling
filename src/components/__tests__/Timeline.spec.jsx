import * as React from "react";
import { shallow } from "enzyme";
import Timeline from "../Timeline";

describe("Timeline", () => {
  const props = {
    flyingTimesList: [
      {
        id: "1234",
        origin: "origin",
        readable_departure: "readable_departure",
        destination: "destination",
        readable_arrival: "readable_arrival",
        departuretime: 1233,
        arrivaltime: 2500
      }
    ]
  };
  const wrapper = shallow(<Timeline {...props} />);
  it("matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
