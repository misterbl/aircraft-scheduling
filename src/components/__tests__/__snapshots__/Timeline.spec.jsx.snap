// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Timeline
    flyingTimesList={
      Array [
        Object {
          "arrivaltime": 2500,
          "departuretime": 1233,
          "destination": "destination",
          "id": "1234",
          "origin": "origin",
          "readable_arrival": "readable_arrival",
          "readable_departure": "readable_departure",
          "timePercentage": 23,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow={1.4270833333333333}
          role="progressbar"
          style={
            Object {
              "width": "1.4270833333333333%",
            }
          }
        />,
        Array [
          <React.Fragment>
            <div
              aria-valuemax="100"
              aria-valuemin="0"
              aria-valuenow={23}
              className="progress-bar bg-success"
              role="progressbar"
              style={
                Object {
                  "width": "23%",
                }
              }
            />
            <div
              aria-valuemax="100"
              aria-valuemin="0"
              role="progressbar"
              style={
                Object {
                  "width": "undefined%",
                }
              }
            />
          </React.Fragment>,
        ],
      ],
      "className": "progress",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1.4270833333333333",
        "nodeType": "host",
        "props": Object {
          "aria-valuemax": "100",
          "aria-valuemin": "0",
          "aria-valuenow": 1.4270833333333333,
          "role": "progressbar",
          "style": Object {
            "width": "1.4270833333333333%",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "23",
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <div
              aria-valuemax="100"
              aria-valuemin="0"
              aria-valuenow={23}
              className="progress-bar bg-success"
              role="progressbar"
              style={
                Object {
                  "width": "23%",
                }
              }
            />,
            <div
              aria-valuemax="100"
              aria-valuemin="0"
              role="progressbar"
              style={
                Object {
                  "width": "undefined%",
                }
              }
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "aria-valuemax": "100",
              "aria-valuemin": "0",
              "aria-valuenow": 23,
              "className": "progress-bar bg-success",
              "role": "progressbar",
              "style": Object {
                "width": "23%",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "aria-valuemax": "100",
              "aria-valuemin": "0",
              "aria-valuenow": undefined,
              "role": "progressbar",
              "style": Object {
                "width": "undefined%",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": Symbol(react.fragment),
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow={1.4270833333333333}
            role="progressbar"
            style={
              Object {
                "width": "1.4270833333333333%",
              }
            }
          />,
          Array [
            <React.Fragment>
              <div
                aria-valuemax="100"
                aria-valuemin="0"
                aria-valuenow={23}
                className="progress-bar bg-success"
                role="progressbar"
                style={
                  Object {
                    "width": "23%",
                  }
                }
              />
              <div
                aria-valuemax="100"
                aria-valuemin="0"
                role="progressbar"
                style={
                  Object {
                    "width": "undefined%",
                  }
                }
              />
            </React.Fragment>,
          ],
        ],
        "className": "progress",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1.4270833333333333",
          "nodeType": "host",
          "props": Object {
            "aria-valuemax": "100",
            "aria-valuemin": "0",
            "aria-valuenow": 1.4270833333333333,
            "role": "progressbar",
            "style": Object {
              "width": "1.4270833333333333%",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "23",
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <div
                aria-valuemax="100"
                aria-valuemin="0"
                aria-valuenow={23}
                className="progress-bar bg-success"
                role="progressbar"
                style={
                  Object {
                    "width": "23%",
                  }
                }
              />,
              <div
                aria-valuemax="100"
                aria-valuemin="0"
                role="progressbar"
                style={
                  Object {
                    "width": "undefined%",
                  }
                }
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-valuemax": "100",
                "aria-valuemin": "0",
                "aria-valuenow": 23,
                "className": "progress-bar bg-success",
                "role": "progressbar",
                "style": Object {
                  "width": "23%",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-valuemax": "100",
                "aria-valuemin": "0",
                "aria-valuenow": undefined,
                "role": "progressbar",
                "style": Object {
                  "width": "undefined%",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": Symbol(react.fragment),
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
